src/explorer.ts

```diff
@@ -348,28 +371,17 @@ class KubernetesNamespaceResource extends KubernetesResource {
     }
 }
 
-class KubernetesServiceFolder extends KubernetesResourceFolder {
-    constructor() {
-        super(kuberesources.allKinds.service);
+class KubernetesPodSelectorFolder extends KubernetesResourceFolder {
+    constructor(readonly kind: kuberesources.ResourceKind, readonly fn: (kubectl: Kubectl) => Promise<kubectlUtils.PodSelector[]>) {
```

Comment:
Rename `fn` to something more meaningful - it has a specific purpose not just a generic callback.

Reply:
fixed.